### PR TITLE
Fix short trajectory plotting

### LINKED ISSUE #76: kinetic enegy distribution from flatparser
Hi everyone,

Thank you for developping this tool!

I'm trying to use it for my simulations carried out using NAMD.
I converted the positions and velocities trajectories into the right format for the flatparser, and it seems working.
I want to check the kinetic energy distribution, and it works UNLESS I specify a filename to display and save the plot...

Here are the code lines I'm using (and work):
```
import physical_validation as pv
import os

ensemble = pv.data.ensemble_data.EnsembleData("NVT", natoms = 219, temperature = 300, volume = 10000)
units = pv.data.unit_data.UnitData(0.001985875, "4.184", "0.1", "0.001", "1", "1", "0.001")
system = pv.data.system_data.SystemData(natoms=219, nconstraints=0,ndof_reduction_tra=0, ndof_reduction_rot=0)
parser =pv.data.flatfile_parser.FlatfileParser()
test = parser.get_simulation_data(units = units, ensemble = ensemble, system = system, dt = 1, position_file = "test_ok.xyz", velocity_file = "vel_ok.xyz", kinetic_ene_file = "kinetic_energy.txt")
pv.kinetic_energy.distribution(test, verbosity = 2, strict = True)
```

If I change the last line for `pv.kinetic_energy.distribution(test, verbosity = 2, filename = "plottest")`, it displays these errors:

`After equilibration, decorrelation and tail pruning, 28.92% (59 frames) of original Kinetic energy remain.
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/shoyas/miniconda2/envs/ChemCoordEnv/lib/python3.7/site-packages/physical_validation/kinetic_energy.py", line 147, in distribution
    **temp_unit=data.units.temperature_str)**
  File "/home/shoyas/miniconda2/envs/ChemCoordEnv/lib/python3.7/site-packages/physical_validation/util/kinetic_energy.py", line 314, in check_mean_std
    **screen=screen)**
  File "/home/shoyas/miniconda2/envs/ChemCoordEnv/lib/python3.7/site-packages/physical_validation/util/plot.py", line 72, in plot
    **_, x, _ = ax.hist(y, r['hist'], **args)**
  File "/home/shoyas/miniconda2/envs/ChemCoordEnv/lib/python3.7/site-packages/matplotlib/__init__.py", line 1438, in inner
    **return func(ax, *map(sanitize_sequence, args), **kwargs)**
  File "/home/shoyas/miniconda2/envs/ChemCoordEnv/lib/python3.7/site-packages/matplotlib/axes/_axes.py", line 6656, in hist
    **m, bins = np.histogram(x[i], bins, weights=w[i], **hist_kwargs)**
  File "<__array_function__ internals>", line 6, in histogram
  File "/home/shoyas/miniconda2/envs/ChemCoordEnv/lib/python3.7/site-packages/numpy/lib/histograms.py", line 792, in histogram
    **bin_edges, uniform_bins = _get_bin_edges(a, bins, range, weights)**
  File "/home/shoyas/miniconda2/envs/ChemCoordEnv/lib/python3.7/site-packages/numpy/lib/histograms.py", line 424, in _get_bin_edges
    **raise ValueError('`bins` must be positive, when an integer')**
ValueError: `bins` must be positive, when an integer`

It seems to always display an error related to the last line of the function involved. For example, in the `/home/shoyas/miniconda2/envs/ChemCoordEnv/lib/python3.7/site-packages/physical_validation/kinetic_energy.py`, the "error" is related to the last line of the `distribution` function.

I really don't understand... 

Could you please help me?

Thank you in advance.

### PR DESCRIPTION
**Pull request by ptmerz**
## Description
Histogram plotting was requiring 150 data points per bin, and didn't prevent having zero bins if the trajectory was shorter. Fixed by explaining the reasoning in a warning, and proceeding with a minimum number of bins if trajectory is shorter.

## Todos
  - [x] Fixes #76

## Status
- [x] Ready to go → #78 should go in first!

### CODE REVIEW

File: physical_validation/util/kinetic_energy.py [hunk 1]
```diff
@@ -155,10 +155,25 @@ def check_distribution(
...
+                    min_num_hist_bins, min_num_data_per_bin, len(kin), min_num_hist_bins
+                )
+            )
+        num_hist_bins = max(min_num_hist_bins, int(len(kin) / min_num_data_per_bin))
+
         data = [
             {
                 "y": kin,
-                "hist": int(len(kin) / 150),
+                "hist": max(5, num_hist_bins),
```
Comment (SimonBoothroyd) [inline physical_validation/util/kinetic_energy.py:176]:
```suggestion
                "hist": max(min_num_hist_bins, num_hist_bins),
```

**Review by SimonBoothroyd — CHANGES_REQUESTED** [review]
LGTM with just a small change!

**Comment by ptmerz** [comment]
Good catch, thanks!

**Comment by ptmerz** [comment]
@SimonBoothroyd, would you have a moment to re-review that? Thanks!

**Review by SimonBoothroyd — APPROVED** [review]
LGTM!

**Pull request merged**